config/config.go

```diff
@@ -90,56 +90,83 @@ func parsePlugins(pluginList []interface{}) ([]*PluginConfig, error) {
 	return plugins, nil
 }
 
-func (c *Config) parseV6Config() error {
-	if exists := c.v.Get("server6"); exists == nil {
-		// it is valid to have no DHCPv6 configuration defined, so no
-		// server and no error are returned
-		return nil
+func (c *Config) getListenAddress(v6 bool) (*net.UDPAddr, error) {
+	ver := 6
+	if !v6 {
+		ver = 4
+	}
+	if exists := c.v.Get(fmt.Sprintf("server%d", ver)); exists == nil {
+		// it is valid to have no server configuration defined, and in this case
+		// no listening address and no error are returned.
+		return nil, nil
 	}
-	addr := c.v.GetString("server6.listen")
+	addr := c.v.GetString(fmt.Sprintf("server%d.listen", ver))
 	if addr == "" {
-		return ConfigErrorFromString("dhcpv6: missing `server6.listen` directive")
+		return nil, ConfigErrorFromString("dhcpv%v: missing `server%d.listen` directive", ver, ver)
 	}
 	ipStr, portStr, err := net.SplitHostPort(addr)
 	if err != nil {
-		return ConfigErrorFromString("dhcpv6: %v", err)
+		return nil, ConfigErrorFromString("dhcpv%d: %v", ver, err)
 	}
 	ip := net.ParseIP(ipStr)
-	if ip.To4() != nil {
-		return ConfigErrorFromString("dhcpv6: missing or invalid `listen` address")
+	if v6 && ip.To4() != nil {
```

Comment:
Why not ip.To16() or checking its length instead of using To4()?

Reply:
`ip.To4()` returns `nil` for all cases where `ip` does not represent a valid IPv4 address, but `ip.To16()` does not return `nil` in the case that it represents a parsed IPv4 address, and similarly I believe length will always be 16 bytes when parsed.  So I think the logic is correct even though it's a little confusing.

That said, I believe there is still a bug here in the case that a bogus address is provided - there needs to be a check that verifies `ip != nil` to determine this, before checking that it's either a valid IPv4 address (!v6), or not a valid IPv4 address (v6).